azurerm/internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go

```diff
@@ -0,0 +1,200 @@
+	return &schema.Resource{
+		Create: resourceArmOrchestratedVirtualMachineScaleSetCreateUpdate,
+		Read:   resourceArmOrchestratedVirtualMachineScaleSetRead,
+		Update: resourceArmOrchestratedVirtualMachineScaleSetCreateUpdate,
+		Delete: resourceArmOrchestratedVirtualMachineScaleSetDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImportThen(func(id string) error {
+			_, err := parse.VirtualMachineScaleSetID(id)
+			return err
+		}, importOrchestratedVirtualMachineScaleSet),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: ValidateVmName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"platform_fault_domain_count": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+				// The range of this value varies in different locations
+				ValidateFunc: validation.IntBetween(0, 5),
+			},
+
+			"single_placement_group": {
+				Type:     schema.TypeBool,
+				Required: true,
+				ForceNew: true,
+			},
+
+			// the VMO mode can only be deployed into one zone for now, and its zone will also be assigned to all its VM instances
+			"zones": azure.SchemaSingleZone(),
+
+			"unique_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmOrchestratedVirtualMachineScaleSetCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Compute.VMScaleSetClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for existing Orchestrated Virtual Machine Scale Set %q (Resource Group %q): %+v", name, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_orchestrated_virtual_machine_scale_set", *existing.ID)
+		}
+	}
+
+	props := compute.VirtualMachineScaleSet{
+		Location: utils.String(location.Normalize(d.Get("location").(string))),
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+		VirtualMachineScaleSetProperties: &compute.VirtualMachineScaleSetProperties{
+			PlatformFaultDomainCount: utils.Int32(int32(d.Get("platform_fault_domain_count").(int))),
+			SinglePlacementGroup:     utils.Bool(d.Get("single_placement_group").(bool)),
```

Comment:
looks like we're not configuring the orchestration mode field here?

Reply:
Yeah... actually there is not a field called `OrchestrationMode` or similar. The only way we can tell whether a VMSS is in an orchestration mode is to check the `resp.VirtualMachineScaleSetProperties.VirtualMachineProfile` is `nil` or not.
Since an orchestrated VMSS will expect the user to add the VM in manually, it should not have a VM template.